micro_dl/cli/image_prediction.py

```diff
@@ -0,0 +1,240 @@
+        fnames = natsort.natsorted(fnames)
+        model_fname = fnames[-1]
+    weights_path = os.path.join(args.model_dir, model_fname)
+
+    # Create image subdirectory to write predicted images
+    pred_dir = os.path.join(args.image_dir, 'predictions')
+    os.makedirs(pred_dir, exist_ok=True)
+    # If saving figures, create another subdirectory to predictions
+    if args.save_figs:
+        target_channel = config['dataset']['target_channels'][0]
+        fig_dir = os.path.join(pred_dir, 'figures')
+        os.makedirs(fig_dir, exist_ok=True)
+
+    # If network depth is > 3 determine depth margins for +-z
+    margin = 0
+    if 'depth' in config['network']:
+        depth = config['network']['depth']
+        if depth > 1:
+            margin = depth // 2
+            metadata_ids['slice_idx'] = aux_utils.adjust_slice_margins(
+                slice_ids=metadata_ids['slice_idx'],
+                depth=depth,
+            )
+    # Get input channel
+    # TODO: Add multi channel support once such models are tested
+    input_channel = config['dataset']['input_channels'][0]
+    assert isinstance(input_channel, int),\
+        "Only supporting single input channel for now"
+    # Get data format
+    data_format = 'channels_first'
+    if 'data_format' in config['network']:
+        data_format = config['network']['data_format']
+    # Load model with predict = True
+    model = inference.load_model(
+        network_config=config['network'],
+        model_fname=weights_path,
+        predict=True,
+    )
+    print(model.summary())
+
+    # Iterate over all indices for test data
+    for time_idx in metadata_ids['time_idx']:
+        for pos_idx in metadata_ids['pos_idx']:
+            for slice_idx in metadata_ids['slice_idx']:
+                im_stack = []
+                # Load +- slice_idx for 3D data
+                for z in range(slice_idx - margin, slice_idx + margin + 1):
+                    meta_idx = aux_utils.get_meta_idx(
+                        frames_meta,
+                        time_idx,
+                        input_channel,
+                        z,
+                        pos_idx,
+                    )
+                    file_path = os.path.join(
+                        args.image_dir,
+                        frames_meta.loc[meta_idx, "file_name"],
+                    )
+                    im = image_utils.read_image(file_path)
+                    # TODO: Add flatfield support
+                    im_stack.append(im)
+                # Stack images
+                im_stack = np.stack(im_stack, axis=2)
+                # Crop image shape to nearest factor of two
+                im_stack = image_utils.crop2base(im_stack)
+                # Flip dimensions if data format is channels first
+                if data_format == 'channels_first':
+                    im_stack = np.transpose(im_stack, [2, 0, 1])
+                # Expand dimensions
```

Comment:
wondering if we could move the functionality b/w lines 160 to 178 to a helper function (read_stack?) in image_utils as this is used here and also in tile_images. To be consistent with tile_images, we could move z_score here and remove it from predict_on_larger_image perhaps?